src/LBToken.sol

```diff
@@ -328,4 +342,22 @@ contract LBToken is ILBToken {
         uint256 id,
         uint256 amount
     ) internal virtual {}
+
+    /// @notice Return if the `_target` contract supports LBToken interface
+    /// @param _target The address of the contract
+    /// @return supported Whether the contract is supported (1) or not (any other value)
+    function _verifyLBTokenSupport(address _target) private returns (uint256 supported) {
+        bytes4 selectorERC165 = IERC165.supportsInterface.selector;
+        bytes4 ILBTokenInterfaceId = type(ILBToken).interfaceId;
+
+        if (_target.code.length > 0)
+            assembly {
+                mstore(0x00, selectorERC165)
+                mstore(0x04, ILBTokenInterfaceId)
+
+                supported := staticcall(30000, _target, 0x00, 0x24, 0x00, 0x20)
+                supported := or(supported, mload(0x00))
```

Comment:
Is there any particular reason to not do `and` here and then checking `supported == true/false` ? Just out of curiosity, I think that's how I would have done it.

Reply:
Yes, as `3 & 5 = 1`, but I think it would be easier to read if we check that both results are exactly 1

Reply:
Oh but `1 | 0 = 1`, I'll check for exact outputs